dlib/pixel.h

```diff
@@ -1024,19 +1115,142 @@ namespace dlib
 
             return(c2);
         }
 
+        /*
+            Calculate HSV from RGB
+            Hue is in degrees
+            Saturation is between 0 and 1
+            Value is between 0 and 1
+        */
+        inline HSX RGB2HSV(COLOUR in)
+        {
+            HSX out;
+            double min, max, delta;
+
+            min = in.r < in.g ? in.r : in.g;
+            min = min < in.b ? min : in.b;
+
+            max = in.r > in.g ? in.r : in.g;
+            max = max > in.b ? max : in.b;
+
+            out.x = max;
+            delta = max - min;
+            if (delta < 0.00001)
+            {
+                out.s = 0;
+                out.h = 0;
+                return out;
+            }
+            if (max > 0.0)
+            {
+                out.s = (delta / max);
+            }
+            else
+            {
+                out.s = 0.0;
+                out.h = NAN;
+                return out;
+            }
+            if (in.r >= max)
+            {
+                out.h = (in.g - in.b) / delta;
+            }
+            else
+            {
+                if (in.g >= max)
+                    out.h = 2.0 + (in.b - in.r) / delta;
+                else
+                    out.h = 4.0 + (in.r - in.g) / delta;
+            }
+
+            out.h *= 60.0;
+
+            if (out.h < 0.0)
+                out.h += 360.0;
+
+            return out;
+        }
+
+        // -----------------------------
+
+        /*
+            Calculate RGB from HSV, reverse of RGB2HSV()
+            Hue is in degrees
+            Saturation is between 0 and 1
+            Value is between 0 and 1
+        */
+        inline COLOUR HSV2RGB(HSX in)
+        {
+            double hh, p, q, t, ff;
+            long i;
+            COLOUR out;
+
+            if (in.s <= 0.0)
+            {
+                out.r = in.x;
+                out.g = in.x;
+                out.b = in.x;
+                return out;
+            }
+            hh = in.h;
+            if (hh >= 360.0)
+                hh = 0.0;
+            hh /= 60.0;
+            i = static_cast<long>(hh);
+            ff = hh - i;
+            p = in.x * (1.0 - in.s);
+            q = in.x * (1.0 - (in.s * ff));
+            t = in.x * (1.0 - (in.s * (1.0 - ff)));
+
+            switch (i)
+            {
+            case 0:
+                out.r = in.x;
+                out.g = t;
+                out.b = p;
+                break;
+            case 1:
+                out.r = q;
+                out.g = in.x;
+                out.b = p;
+                break;
+            case 2:
+                out.r = p;
+                out.g = in.x;
+                out.b = t;
+                break;
+            case 3:
+                out.r = p;
+                out.g = q;
+                out.b = in.x;
+                break;
+            case 4:
+                out.r = t;
+                out.g = p;
+                out.b = in.x;
+                break;
+            case 5:
```

Comment:
```suggestion
```
The original code had this extra case, which seems completely useless. Or am I not seeing something?

Reply:
Yeah that doesn't do anything.